### ISSUE TITLE
Support maximum repetition count

### ISSUE THREAD

**Issue by josephcsible** [feuerbach/regex-applicative#38]
Consider the regex `\d{,6}`. I'd like this library to contain a function `replicateAlt` that lets me implement this regex as `replicateAlt 6 digit`. One possible implementation would be a duplicate of [`replicateM`](https://hackage.haskell.org/package/base-4.12.0.0/docs/src/Control.Monad.html#replicateM), but with `<|> pure []` on the end of the recursive case.

If you like this idea, I can PR it, but I'm not sure which module you want it in.

**Comment by feuerbach** [comment]
Does [this](https://hackage.haskell.org/package/parser-combinators-1.2.0/docs/Control-Applicative-Combinators.html#v:count-39-) do what you want?

I'd accept a PR explaining (in the haddocks) how to do this for others who may be looking for the same thing.

**Comment by josephcsible** [comment]
Yes, that's what I need for now. Although it looks like they only have a greedy one, so if I instead needed the lazy `\d{,6}?`, I'd be right back to needing to roll my own.

**Comment by feuerbach** [comment]
Ah, good point. In that case it may be worth defining our own function.

**Comment by stuart-little** [comment]
I dropped in intending to ask this in a separate issue, but this seems like a good spot: 

What *is* the best way to quantify numerically? I know you can cobble something together out of what's already there. For instance, to match exactly `n` digits I can do 

```
ghci> dig :: RE Char String; dig = pure <$> psym isDigit
ghci> <string> =~ (foldr1 (<>) $ replicate n dig)
```
But is this more or less it? Or is there some [count](https://hackage.haskell.org/package/base-4.15.0.0/docs/Text-ParserCombinators-ReadP.html#v:count)-style combinator already built in, that I'm missing in the docs?

**Comment by UnkindPartition** [comment]
Have you seen the `count` combinator linked above?

**Comment by stuart-little** [comment]
> Have you seen the `count` combinator linked above?

My apologies: I did see it, but did not understand the context.. 

I hadn't realized it applies to *all* instances of alternative. I thought it was more like the `count` I linked to, which was specific to `ReadP`.

Thank you!
